.github/workflows/ci.yml

```diff
@@ -78,6 +79,16 @@ jobs:
     steps:
     - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+        key: ${{ runner.os }}-cargo-test-CI2-${{ hashFiles('**/Cargo.toml') }}
```

Comment:
I will remove the `-test-CI2-` when the PR going to merge.

Reply:
(Commenting here so this turns into a review comment and we make sure to do it before I approve the PR)

Reply:
If we change the key below, make sure to update it here too:

> If we're just caching global Cargo files, is including the hashes of `Cargo.toml` files in the key unnecessary? Presumably the final version of this key could just be `${{ runner.os }}-cargo`?

Reply:
Sure